uniffi_bindgen/src/scaffolding.rs

```diff
@@ -53,9 +53,10 @@ mod filters {
             Type::Float64 => "f64".into(),
             Type::Boolean => "bool".into(),
             Type::String => "String".into(),
-            Type::Enum(name) | Type::Record(name) | Type::Object(name) | Type::Error(name) => {
+            Type::Enum(name) | Type::Record(name) | Type::Error(name) => {
                 name.clone()
             }
+            Type::Object(name) => format!("std::sync::Arc<{}>", name),
```

Comment:
What this is saying is saying is essentially, anywhere where the UDL says we pass an object instance, the Rust code should accept an `Arc<T>`. We don't actually use it anywhere directly right now, but in the future if we want to accept an object instance as an argument to a function, this would imply that the function accepts an `Arc<T>`.

How to indicate that `[ByRef] Object` arguments accept an `&T` is not yet clear; right now `[ByRef]` causes UniFFI to just append an `&` to the type defined here, so `[ByRef] Object` would be translated into `&Arc<T>`, which is not what we want.